### PR TITLE
Include cstring and cstddef for C++ code

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Include cstring and cstddef for C++ code

### CODE REVIEW

File: macro_utils_h_generator/macro_utils.tt [hunk 1]
```diff
@@ -23,8 +23,13 @@
 #ifndef MACRO_UTILS_H
 #define MACRO_UTILS_H
 
+#ifdef __cplusplus
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: >__cplusplus [](start = 7, length = 11)

consider then adding #ifdef __cplusplus / extern "C" { #endif to the potential function declarations/definitions that can arise form this file.

**Comment by anporumb** [comment]
> #define MU_DEFINE_MICROMOCK_ENUM_TO_STRING(type, ...) MICROMOCK_ENUM_TO_STRING(type, MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT_AS_WIDESTRING, __VA_ARGS__));

can this be removed?

---
Refers to: inc/azure_macro_utils/macro_utils.h:12777 in e9bcb7f. [](commit_id = e9bcb7f23330f860dada83d1f75ebda294ee5367, deletion_comment = False)

File: macro_utils_h_generator/macro_utils.tt (hunk 1 above)
Comment (dcristoloveanu) [reply macro_utils_h_generator/macro_utils.tt]: Added

---
In reply to: [280658747](https://github.com/Azure/azure-macro-utils-c/pull/13#discussion_r280658747) [](ancestors = 280658747)

**Comment by dcristoloveanu** [comment]
> #define MU_DEFINE_MICROMOCK_ENUM_TO_STRING(type, ...) MICROMOCK_ENUM_TO_STRING(type, MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT_AS_WIDESTRING, __VA_ARGS__));

Will remove it and file a PR to add it to Micromock C++ since apparently it is still used by some old IoT SDK tests

---
In reply to: [488920095](https://github.com/Azure/azure-macro-utils-c/pull/13#issuecomment-488920095) [](ancestors = 488920095)

---
Refers to: inc/azure_macro_utils/macro_utils.h:12777 in e9bcb7f. [](commit_id = e9bcb7f23330f860dada83d1f75ebda294ee5367, deletion_comment = False)

**Pull request merged**